src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -135,12 +147,14 @@ void IEnumerator.Reset()
 
         internal class WhereEnumerableIterator<TSource> : Iterator<TSource>
```

Comment:
It probably should be. This class has been `internal` since it was brought into the repository, and I made the comparable classes introduced here, like `SelectEnumerableIterator`, `internal` to match that. If there's no good reason for these to be `internal` then I think they should all be made `private` to give a better ability to eyeball that they're correct. While I made a few fields `readonly` for similar ability to assure oneself of correctness at a glance, that was only because I'd changed the degree that invariant was depended on (if `_predicate` changed, part of my logic was flawed).
In other words, I think you're right, but I think I shouldn't change it right here were it's not relevant.